client/client_test.go

```diff
@@ -518,6 +518,35 @@ func TestAddTargetToTargetRoleByDefault(t *testing.T) {
 		[]string{data.CanonicalTargetsRole})
 }
 
+// Adding a target ignores the role of the target object, and goes by the
+// roles passed to AddTarget (or the default targets role, if no roles are passed)
+func TestAddTargetIgnoresTargetRole(t *testing.T) {
+	ts, _, _ := simpleTestServer(t)
+	defer ts.Close()
+
+	repo, _ := initializeRepo(t, data.ECDSAKey, "docker.com/notary", ts.URL, false)
+	defer os.RemoveAll(repo.baseDir)
+
+	targetObj, err := NewTarget("latest", "../fixtures/root-ca.crt")
+	assert.NoError(t, err, "error creating target")
+	targetObj.Role = "targets/whosits"
```

Comment:
Heh, sorry, this is a misremembered [character from A Wrinkle In Time](https://en.wikipedia.org/wiki/A_Wrinkle_in_Time#Mrs_Whatsit) (I got the name wrong).  Will change it to `targets/whoisit` :)